lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,173 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
         if clip_path:
             self.patch.set_clip_path(*clip_path)
         self.patch.draw(renderer)
+
+
+class TickedStroke(AbstractPathEffect):
+    """
+    A line based PathEffect which draws a path with a ticked style.
+    This line style is frequently used to represent constraints in
+    optimization.  The ticks indicate that one side of the line is
+    invalid.
+
+    The spacing, length, and angle of ticks can be controlled.
+
+    This line style is sometimes referred to as a hatched line, but
+    in matplotlib, hatching typically refers to filling an area with
+    a line pattern.
+
+    """
+
+    def __init__(self, offset=(0, 0),
+                 spacing=10.0, angle=45.0, length=np.sqrt(2),
+                 flip=False, **kwargs):
+        """
+        Parameters
+        ----------
+        offset : pair of floats
+            The offset to apply to the path, in points.
+        spacing : float
+            The spacing between ticks in points.
+            Default is 10.0.
+        angle : float
+            The angle between the path and the tick in degrees.  The angle
+            is measured as if you were an ant walking along the curve, with
+            zero degrees pointing directly ahead, 90 to your left, 180
+            behind you, and 270 to your right.  If the curve was
+            generated by contour, any constraint in negative null form
```

Comment:
This last sentence is too jargony.  I have no idea what "negative null form" is of how it relates to a g(x) from the rest of this documentation.  I can guess its related to "constraints in optimization", but I don't think this sentence adds anything for the general user.

Reply:
How is 4b0faa27f below?

Reply:
I still don't know what you mean.  If you want to have domain-specific comments like this, I'd put them in the example where we can see what you are talking about.  I appreciate this is why you wrote the style, but I think its more generally usable, and would recommend not confusing the user.

Reply:
I agree that this is of more general use.

The third example in the demo set is an example of using this in a the specialized context.  You must stay on the 'clean' side of the lines (upper-right domain), but you would like to get to the bottom of the circles towards the bottom-left corner.  The solution is where the two straight-line constraints cross.  I will add some discussion to that demo.

I didn't see any way to link to the demo from the documentation.  Is there a way to make that link -- or do we rely on the user cruising the demo pages?

I'm trying to make this comment as 'footnote like' as possible -- that way it can be ignored by everyone who doesn't need it, but the information is still there for those that do.

In this category of problem (which is actually quite broad), you really care which side of the line the ticks appear.  You may produce a plot with 5-20 such ticked lines on it.  You don't have time to manually check them all -- and your intuition may not be easily correct.

Matlab makes it worse because the contour algorithm (contourc) makes no guarantee of the orientation of the contour lines.  I've developed 'ocontourc' on Matlab to go with a [hatched-line drawing routine there](https://www.mathworks.com/matlabcentral/fileexchange/29121-hatched-lines-and-contours) to provide oriented contours.  Matlab's contourc is the worst-case scenario because the orientation of the hatches becomes effectively random -- small changes to your program can cause the direction to flip.

In constrained optimization problems, it is convention to put all constraints into what we call negative null form.  Constraint functions are called g(), the design vector is x, negative null form means that negative (and zero) values satisfy the constraint.  I.e. g(x)<=0.

My intent with this footnote is to tell this category of users that if they choose angle (0 180), that they will get their desired result.  They can have confidence in this, it is by design, and it will not randomly change on them.

Reply:
You can specify links to examples and tutorials with the `:doc:` directive.  

```
See :doc:`/tutorials/intermediate/constrainedlayout_guide`
```

Again, I've used constrained optimizations quite a few times over the years, and still consider what you wrote in the docstring jargon.  No doubt there is some subset of folks who have standardized plots that use `g()` to  refer to the constraint, but that's *not* universal enough to just put in the middle of a docstring.  

I'd add a sentence at the top that says something like "often used to depict constrained regions of a minimization space, see `:doc:/examples/contours/your-example`" but leave this out of the description of "angle".  If you want to specify that the ticks will always be on the correct side of the line, that's useful general information.

Reply:
I've removed the reference to g() in 4b0faa2 and have moved the comment to the Demo in 5d45476.

I brought it up again in the discussion to provide context -- I wasn't arguing to put it back in.  Your comments about this being specialized or esoteric made me think some additional explanation would be helpful.  I don't know your background, so I was trying to cover my bases.

I think it may be worthwhile to split the demo into two -- keep the first two demos where they are as a generic patheffect demo.  The other as an example for constrained optimization problems.

A user looking for this capability probably won't find it under misc/ but would be more likely to find it near similar demos in images_contours_and_fields.

Reply:
No that seems great.  Thanks for your work on this.I'll take a quick look now